### ISSUE TITLE
Missing coverage for mixin class when testing derived class?

### ISSUE THREAD

**Issue by BrendanDrew** [pytest-dev/pytest-cov#198]
I'm not sure if this is a bug or not.

Scenario:
I have an implementation of Levenberg-Marquard that I'm using to "invert" a function. Because I am interested in the broad category of all such functions (not just one particular form), I thought I'd be "clever" and implement this particular portion as a mixin that can be freely inherited by client classes. Unfortunately, the mixin itself is not directly testable (it requires that client classes provide certain key pieces of information, such as gradients and the forward operation which I am attempting to invert). 

When I look at test coverage reports, I see clear indication that I have run tests which exercise calls to the mixin's methods via its concrete subclasses / clients (I know because I currently have tests that fail after making these calls -- d'oh!). What is confusing to me is that when I look at the coverage reports, I see the entire mixin being marked as uncovered. 

What are reasonable expectations in this case? I can imagine an implementation that, were a test to fail, pytest / pytest-cov may drop any information related to that test (such as coverage) and that once those tests pass, I /should/ see the coverage on the base class? If not, what are the "right" ways to go about testing classes which are not directly instantiatable (e.g. mixins, traditional OO abstract base classes, ...) in a way that produces true estimates of test coverage for those classes? 

Thanks!

**Comment by blueyed** [comment]
1. Is it imported before coverage starts?
2. Are there also passing tests using the mixin?

**Comment by BrendanDrew** [comment]
1. No (this is a pet-peeve of mine with how the import system in python works -- it makes reasoning about the system under test really difficult because the imports alter the state of the system in not-obvious ways). 
2. I had gone so far as to artificially pass a test to see if I could rule out this possibility before asking the question. 

Fortunately, I managed to find what was going wrong. My apologies for the false alarm, this was clearly a PEBKAC (problem exists between keyboard and chair) situation. 

I am not a native pythonista, so I'd missed that the mixin code wasn't being called due to the method resolution order that had been chosen by the interpreter (d'oh!), so this was absolutely correct behavior on pytest-cov / coverage.py part. Now that I have fixed this particular idiocy, I see the coverage being reported cleanly. 

Thanks! Tools like this make python development a joyful experience for me. It's also been very gratifying that I've been able to share these tools with colleagues and see them start to rack up meaningful wins because of the visibility that pytest-cov has helped create.
